tests/integration/models/device.spec.coffee

```diff
@@ -1028,6 +1029,28 @@ describe 'Device Model', ->
 					m.chai.expect(_.find(result, { name: 'A' }).value).equal('a')
 					m.chai.expect(_.find(result, { name: 'B' }).value).equal('b')
 
+		describe 'balena.models.device.getSupervisorTargetState()', ->
+
+			it 'should be rejected if the device does not exist', ->
+				promise = balena.models.device.getSupervisorTargetState('asdfghjkl')
+				m.chai.expect(promise).to.be.rejectedWith('Device not found: asdfghjkl')
+
+			it 'should reflect the device\'s name', ->
+				balena.models.device.getSupervisorTargetState(@device.id).then (state) ->
+					m.chai.expect(state.local.name).to.be.a('string')
+
+			it 'should reflect the device\'s apps', ->
+				balena.models.device.getSupervisorTargetState(@device.id).then (state) ->
```

Comment:
I think that the issue is that you need `=>` instead of `->` so that the `this` is still the same inside the promise handler:
```suggestion
				balena.models.device.getSupervisorTargetState(@device.id).then (state) =>
```